### DISCUSSION TITLE
Ultra HDR support

### DISCUSSION THREAD

**Discussion by laureanorp** [ente/ente#779] [Enhancements]
**Describe the feature**
Hello! First time Ente user, coming from almost a decade of Google Photos. I trued to search in the issues for something mentioning HDR or Ultra HDR in photos, couldn't find anything. So, basically, it would be interesting to have Ultra HDR support when viewing photos in Ente. More and more platforms like Instagram are being updated tobsupport viewing this kind of format. I can't find a lot of documentation about it, but here is a link about it:

https://www.androidpolice.com/android-14-ultra-hdr-hands-on/

I know, since it's probably dependant on Android 14 and also requires an HDR capable device, it won't be up in the priority list. But just wanted to leave it here as an idea :)

**Comment by Gavlitskii** [comment]
Totally agree. Especially as an iOS user, most of the photos I take clearly cannot be viewed in HDR properly in Ente which is a dealbreaker for me. It is clearly visible even to a non tech audience. Plus HDR photos are now default on iPhones afaik 

HDR support should be a very high priority since it is stopping a lot of people I know from switching (including online)

I hope we’ll get a feedback from Ente soon!

**Comment by twizzlebizzle** [comment]
Just wanted to bump this and throw my hat in the ring.  Just made the switch and its the first thing I noticed.  I am keeping offline google photos just for this currently.

**Comment by CiottiGiorgio** [comment]
I also think supporting this is important. Until it's supported, I'm going to have to keep a separate app around, which in my case is Google Photo.

**Comment by europrobe** [comment]
Bump. I don't want my photos to look dull, and I don't want to have to explicitly download photos one by one from Ente and then open them in Google Photos.

This is a dealbreaker for me. (I'm on Android)

**Comment by twizzlebizzle** [comment]
@vishnukvmd @enteio Can we get any commitment on this feature please?  It's pretty common on all new android phones and does improve the viewing experience.

**Reply by twizzlebizzle** [reply]
@enteio @vishnukvmd 
Sorry tagging you again! This feature is so important to me and others, it's quite a common features is there anything in development?

**Comment by Zarkorix** [comment]
Also a very important feature for me too. If it helps to take a look at, it has been implemented in other FOSS packages already, such as: https://github.com/FossifyOrg/Gallery.

**Comment by Reeceeboii** [comment]
+1 with everyone above. Really want my photos/videos to be displayed at their best without needing to rely on a separate app to view them

**Comment by zilexa** [comment]
https://github.com/ente-io/ente/discussions/7560 should be merged into this one. Adding 5 votes.

**Comment by zilexa** [comment]
This may be the reason why it's impossible for Ente to implement UltraHDR:
https://github.com/flutter/flutter/issues/127852#issuecomment-2764670878

But higher bit depth might be possible.

**Comment by bit-whisperer** [comment]
bump, I just noticed with ente that Ultra HDR is not supported. If Ultra HDR is supported on ente Photos, then this would be a no brainer to just stop using Google Photos completely

**Reply by bit-whisperer** [reply]
okay I just went into the web app for ente on my phone (I am using Vivaldi) and HDR is being passed through correctly on my Pixel 10. It seems like, its not that ente has no way to read HDR, it is just the android app itself has no way to process the HDR it seems

**Reply by zilexa** [reply]
9

> okay I just went into the web app for ente on my phone (I am using Vivaldi) and HDR is being passed through correctly on my Pixel 10. It seems like, its not that ente has no way to read HDR, it is just the android app itself has no way to process the HDR it seems

I believe Ente is a Flutter app. The Flutter framework doesn't do UltraHDR and isn't even working on supporting it. See my comment above. Nothing Ente can do about that. If you dive in deeper, Flutter may support wide color gamut. I think that's a step forward. Not sure what to expect from it.

**Reply by vancura** [reply]
That's my understanding too. The desktop application (at least macOS, which I tested) supports HDR images correctly (JPEGs with gainmaps, JPEG/XL, and AVIF). In the iOS app, I get artifacts like these:

<img width="1290" height="2796" alt="9FA081EC-AAA3-4C4A-B8F5-E1B973A54E70" src="https://github.com/user-attachments/assets/9802934d-c756-4bb1-9c67-c3f567a16ad9" />

<img width="1290" height="2796" alt="E8ECCDAC-DCF9-4D74-AE25-BECBFF156FDB" src="https://github.com/user-attachments/assets/86e208de-7944-4d79-af04-bd962f4c2b09" />

It seems this is some debug spectrum mapped to above-the-headroom colors, IMHO.
